src/planet-a/Active/SysPlanetA.cpp

```diff
@@ -210,9 +245,123 @@ void SysPlanetA::update_geometry()
                                  MeshVisualizer3D::Normal{})
                 .setIndexBuffer(planet.m_indxBufGL, 0,
                                 MeshIndexType::UnsignedInt)
-                .setCount(planet.m_planet.calc_index_count());
+                .setCount(rPlanetGeo.calc_index_count());
+        }
+
+        if (m_debugUpdate.triggered() || true)
+        {
+            planet_update_geometry(ent);
+        }
+    }
+}
+
+void SysPlanetA::planet_update_geometry(osp::active::ActiveEnt planetEnt)
+{
+    auto &rPlanetPlanet = m_scene.reg_get<ACompPlanet>(planetEnt);
+    auto const &planetTf = m_scene.reg_get<ACompTransform>(planetEnt);
+    auto const &planetActivated = m_scene.reg_get<ACompActivatedSat>(planetEnt);
+
+    // TODO: de-spaghettify systems. make systems entirely stateless
+    osp::universe::Universe const &uni = m_scene.dynamic_system_find<SysAreaAssociate>().get_universe();
+
+    Satellite planetSat = planetActivated.m_sat;
+    auto &planetUComp = uni.get_reg().get<universe::UCompPlanet>(planetSat);
+
+    PlanetGeometryA &rPlanetGeo = *(rPlanetPlanet.m_planet);
+
+    // Temporary: use the first camera in the scene as the viewer
+    ActiveEnt cam = m_scene.get_registry().view<ACompCamera>().front();
+    auto const &camTf = m_scene.reg_get<ACompTransform>(cam);
+
+    // Set this somewhere else. Radius at which detail falloff will start
+    // This will essentially be the physics area size
+    float viewerRadius = 64.0f;
+
+    // camera position relative to planet
+    Vector3 camRelative = camTf.m_transform.translation()
+                        - planetTf.m_transform.translation();
+
+    using Magnum::Math::max;
+    using Magnum::Math::pow;
+    using Magnum::Math::sqrt;
+
+    // Ratio between an icosahedron's edge length and radius
+    static const float sc_icoEdgeRatio = sqrt(10.0f + 2.0f * sqrt(5.0f)) / 4.0f;
```

Comment:
This should be constexpr.

Is Magnum::Math::sqrt not constexpr?

If not, then I think there's sufficient justification for us to define a constexpr_sqrt function that uses doubles for the calculation, with a naive implementation (It happens at compile time, so who cares if it's fast, I guess) to use when all the constants are known at compile time.

Reply:
Unfortunately, there isn't a constexpr sqrt in magnum nor std. The simplest solution possible that will work for sure is to just use the calculated result (0.9510565162). It is extremely unlikely that this equation will ever change. I'm not sure how reliable a constexpr float sqrt function will be.

Reply:
No more or less reliable than a hand-specified outcome, i suppose.